test/integration/query/deletes.js

```diff
@@ -82,5 +84,61 @@ module.exports = function (knex) {
           );
         });
     });
+
+    describe('Delete with join', function () {
+      it('should handle basic delete with join', async function () {
+        const query = knex('test_table_two')
+          .join('accounts', 'accounts.id', 'test_table_two.account_id')
+          .where({ 'accounts.email': 'test3@example.com' })
+          .del();
+        if (isSQLite(knex) || isPostgreSQL(knex) || isOracle(knex)) {
```

Comment:
Do you see downside of using this approach for PostgreSQL - https://www.postgresqltutorial.com/postgresql-delete-join/?

Reply:
@kibertoad  I did not think about it since it has a different syntax.
The best solution for PostgreSQL I think would be to add direct support for `USING` syntax like a `.using` method, it can use the same interface with the current `.join`.
Overall, internally transforming from `.join` might be convenient and helpful too. Among downsides, maybe a small confusion from query transformation.

Reply:
Fair. Converting joins to non-joins is likely to be confusing.

Reply:
In this case it would seem to be quite trivial crossdb compatibility thing to implement... though, maybe native syntax should be allowed too. 

Also tests for error cases where people tries to create .join queries for unsupported databases seem to be missing.

Reply:
@Egor-Koldasov Could you add these in a follow-up PR?

Reply:
@OlivierCavadenti Delete join support with `USING` syntax for PG at this point is probably the largest gap in functionality still missing for 1.0.0. If you'll have some spare time, any chance you could take a look at it?

Reply:
@kibertoad yeah I can work on it soon. As I understand, and to be sure, we go on a syntax from :

```
knex
	.select('xyz')
	.from('my_table')
	.leftJoin('users', 'users.user_id', 'my_table.user_id')
        .where('my_table.criteria', '<', 10)
	.delete();
```

to in PG (no hidden converting left join to using ?):

```
knex
	.select('xyz')
	.from('my_table')
	.using('users')
        .where('my_table.criteria', '<', 10)
        .andWhere('users.user_id', 'my_table.user_id')
	.delete(); 
```

Reply:
Yes, exactly!